sky/serve/autoscalers.py

```diff
@@ -160,7 +191,19 @@ def update_version(self, version: int,
         self.scale_down_consecutive_periods = int(
             downscale_delay_seconds /
             constants.AUTOSCALER_DEFAULT_DECISION_INTERVAL_SECONDS)
-        self.bootstrap_done = False
+
+        # Recalculate target_num_replicas based on QPS.
+        new_qps_target_num_replicas = math.ceil(
+            len(self.request_timestamps) / self.qps_window_size /
+            self.target_qps_per_replica
+        ) if self.target_qps_per_replica is not None else spec.min_replicas
```

Comment:
nit: wrap this to a private function (start with leading `_`) and replace here & `_set_desired_num_replicas`